cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/OptimizationOptions.java

```diff
@@ -17,6 +17,8 @@
   private final Set<Integer> _excludedBrokersForReplicaMove;
   private final boolean _isTriggeredByGoalViolation;
   private final Set<Integer> _requestedDestinationBrokerIds;
+  // _onlyMoveImmigrantReplicas is set to true only when modifying topic's replication factor.
```

Comment:
If that is the case, should we call this variable `isChangingReplicationFactor` and then explain what changing replication factor implies?

Reply:
My concerns is this may be changed, i.e some other operations we want to add have the same restriction.

Reply:
Then can we move this assertion from here to `KafkaCruiseControl#updateTopicConfiguration(...)`?